pkg/crc/cluster/cluster.go

```diff
@@ -138,6 +139,41 @@ func StopAndRemovePodsInVM(sshRunner *ssh.SSHRunner) error {
 	return errors.RetryAfter(2, stopAndRemovePods, 2*time.Second)
 }
 
+func AddProxyConfigToCluster(oc oc.OcConfig, proxy *network.ProxyConfig) error {
+	cmdArgs := []string{"patch", "proxy", "cluster", "-p",
+		fmt.Sprintf(`{"spec":{"httpProxy":"%s", "httpsProxy":"%s", "noProxy":"%s"}}`, proxy.HttpProxy, proxy.HttpsProxy, proxy.NoProxy),
+		"-n", "openshift-config", "--type", "merge"}
+
+	if err := waitForOpenshiftAPIServer(oc); err != nil {
+		return err
+	}
+	_, stderr, err := oc.RunOcCommand(cmdArgs...)
+	if err != nil {
+		return fmt.Errorf("Failed to add proxy details %v: %s", err, stderr)
+	}
+	return nil
+}
+
+func AddProxyToKubeletAndCriO(sshRunner *ssh.SSHRunner, proxy *network.ProxyConfig) error {
+	proxyTemplate := `[Service]
+Environment=HTTP_PROXY=%s
+Environment=HTTPS_PROXY=%s
+Environment=NO_PROXY=%s`
+	defaultNoProxyForCluster := ".cluster.local,.svc,10.128.0.0/14,172.30.0.0/16"
+	noProxy := fmt.Sprintf("%s,%s", defaultNoProxyForCluster, proxy.NoProxy)
+
+	p := fmt.Sprintf(proxyTemplate, proxy.HttpProxy, proxy.HttpsProxy, noProxy)
+	_, err := sshRunner.Run(fmt.Sprintf("cat <<EOF | sudo tee /etc/systemd/system/crio.service.d/10-default-env.conf\n%s\nEOF", p))
+	if err != nil {
+		return err
+	}
+	_, err = sshRunner.RunPrivate(fmt.Sprintf("cat <<EOF | sudo tee /etc/systemd/system/kubelet.service.d/10-default-env.conf\n%s\nEOF", p))
```

Comment:
why `RunPrivate` ? I don't think there's anything secret in that command?

Reply:
@cfergeau the proxy variable which have password are going to be exposed if we are not use `RunPrivate` on this.